### PR TITLE
Feature/set gender

### PR DESCRIPTION
**Pull request by henrikstranneheim**
…en Active_parameter.pm and Gender.pm

- Added new subs and removed others
- Updated and added tests

### This PR adds | fixes:

- See above

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Active_parameter.pm [hunk 1]
```diff
@@ -77,6 +80,50 @@ BEGIN {
     };
 }
 
+sub add_gender {
+
+## Function : Add sample_ids to list of genders of the current analysis
```
Comment (jemten) [inline lib/MIP/Active_parameter.pm:85]:
```suggestion
## Function : Add sample_id to list of genders of the current analysis
```

File: lib/MIP/Active_parameter.pm [hunk 2]
```diff
@@ -1790,11 +1926,11 @@ sub set_gender_sample_ids {
...
-        1      => \&_set_male_gender,
-        2      => \&_set_female_gender,
-        female => \&_set_female_gender,
-        male   => \&_set_male_gender,
-        other  => \&_set_other_gender,
+        1      => q{males},
+        2      => q{females},
+        female => q{females},
+        male   => q{males},
+        other  => q{others},
```
Comment (jemten) [inline lib/MIP/Active_parameter.pm:1933]: nice change from dispatch table to a generic sub 👍

File: lib/MIP/Parse/Gender.pm [hunk 3]
```diff
@@ -441,25 +445,50 @@ sub update_gender_info {
```
Comment (jemten) [inline lib/MIP/Parse/Gender.pm:445]: maybe we could play with a ternary operator here.
```Perl
$y_read_count > $MALE_THRESHOLD ? $gender = q{male} : $gender = q{female};
$log->info(qq{Found $gender according to fastq reads});
...
```
And then skip the if - else statement. This is more of an idea, the code is perfectly readable (if not more ) as is.

File: lib/MIP/Active_parameter.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Active_parameter.pm:1933]: Thx! It felt like an overkill with a dispatch and I also wanted to reuse the function elsewhere

File: lib/MIP/Parse/Gender.pm (hunk 3 above)
Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm:447]: Good idea! I made a suggestion

Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm:447]: Please have a look

Comment (jemten) [reply lib/MIP/Parse/Gender.pm:447]: Looks good! More compact and no readability lost. Whats not to like 😄

Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm:447]: 👍

**Pull request merged**